src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -21,27 +21,40 @@ public class GraphQLHttpMiddleware<TSchema>
         private const string GraphQLContentType = "application/graphql";
 
         private readonly RequestDelegate _next;
-        private readonly PathString _path;
+        private readonly GraphQLHttpMiddlewareOptions _options;
 
-        public GraphQLHttpMiddleware(RequestDelegate next, PathString path)
+        public GraphQLHttpMiddleware(RequestDelegate next, GraphQLHttpMiddlewareOptions options)
         {
             _next = next;
-            _path = path;
+            _options = options;
         }
 
         public async Task InvokeAsync(HttpContext context)
         {
-            if (context.WebSockets.IsWebSocketRequest || !context.Request.Path.StartsWithSegments(_path))
+            if (context.WebSockets.IsWebSocketRequest || !context.Request.Path.StartsWithSegments(_options.Path))
             {
                 await _next(context);
                 return;
             }
 
+            // Authorize request
+            if (!await HttpAuthorizationHelper.AuthorizeAsync(context, _options.AuthorizationPolicyName))
```

Comment:
Ok, I understand this now. So `AuthorizationPolicyName` is like a default policy for the entire API. When you call Microsofts `AddAuthorization` function you can already set a default policy there:

```
services.AddAuthorization(options => options.DefaultPolicy = "FooPolicy");
```

I think you should take a look at the code from my PR https://github.com/graphql-dotnet/authorization/pull/13 to be able to add auth policies to individual graph fields or graph types.

Reply:
I think `UseAuthorization()` actually applies the `DefaultPolicy` above. So there is already middleware that will achieve this built into ASP.NET Core.

Reply:
Confirmed it doesn't exist: https://github.com/aspnet/Security/issues/1734.

Reply:
You're absolutely right, I must have imagined it. Looks like they are planning to add a middleware to do this though. If we can't wait for Microsoft to implement it, perhaps it should be a separate `UseDefaultAuthorizationPolicy` middleware.

Either way, we should still not add `AuthorizationPolicyName` to `GraphQLHttpMiddlewareOptions`. Instead we should use `DefaultPolicy` in the authorization options.

Reply:
I thought about using the `DefaultPolicy`, but what if you want to use the default policy for MVC and a different policy for your GraphQL endpoint? With the `AuthorizationPolicyName` you're able to specify whichever policy you want to use.

Reply:
If there was a generic middleware, I guess you could do something like this:

``` csharp
app.Map("/graphql", branch =>
{
    branch.UseAuthorization(policy: "IsAuthenticated");
    branch.UseGraphQL<TestSchema>("");
});
```

Reply:
You can run multiple apps in a single ASP.NET Core application with separate DI containers, see [blog post](https://www.strathweb.com/2017/04/running-multiple-independent-asp-net-core-pipelines-side-by-side-in-the-same-application/). Not well known you can do this.